### PR TITLE
Refactor Diderot CLI to use python's "click" package

### PR DESCRIPTION
**Pull request by arturts**
[click](https://click.palletsprojects.com/)

[pre-release](https://github.com/diderot-edu/diderot-cli/releases/tag/v0.1.0-rc)

To install the release locally using pip: 
```sh
pip install git+https://github.com/diderot-edu/diderot-cli.git@v0.1.0-rc
```

## Creating distribution
Requirement: pip install wheel
Under root, type `make dist`
This creates `dist/diderot_cli-0.1.0-py3-none-any.whl`
You can pip install this file `pip install dist/diderot_cli-0.1.0-py3-none-any.whl`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,13 +10,14 @@ We thank William Paivine for his initial work on this project.
 
 ## Installation
 
-To install the CLI, clone this repo. The CLI depends only on python's `requests` package, and requires python 3.7 or higher.
+To install the CLI, either clone this repo and run `make install`, or download the pre-packaged `wheel` file and run `pip install path/to/didert_cli-VERSION-[...].whl`. It requires python 3.7 or higher.
+The dependencies are automatically installed if not present.
 
-To install the dependencies, use `pip install -r requirements.txt`.
+If you do not wish to install this package globally, you can create a virtualenv, install the dependencies manually wih `pip install -r requirements.txt` (may include development dependencies as well) and use the `diderot`, `diderot_admin` or `diderot_student` scripts at the root of this project.
```
Comment (rohany) [inline README.md:16]: /s/wih/with

**Review by rohany — COMMENTED** [review]
Cool refactor! I've never used click before, so I'm not able to tell if it's right or not, but the code looks simpler than the original argparse based approach. I've also never published a python package, so IDK if that code is correct either :)

**Comment by arturts** [comment]
For the package publishing part I was following click's tutorial, I didn't know anything about it either ¯\_(ツ)_/¯ 
Took a few tries, but I could manage to run in two different machines (MacBook air 2018 I think and an M1 Mac mini) by running the pip install <url> command. 
This way we can add development-only packages to requirements.txt but the user only needs to have the ones defined in setup.py

**Comment by rohany** [comment]
Wow, being able to pip install diderot sounds pretty cool :D

File: README.md (hunk 1 above)
Comment (arturts) [reply README.md]: 🦅

**Comment by umutacar** [comment]
Some refinements:
* env var utility is nice.  let's add to the README
* it seems that we need to create a diderot-cli namespace for all packages used here (because otherwise python cannot resolve conflicts)

**Comment by umutacar** [comment]
Time to try this out :)

**Pull request merged**